lib/ansible/plugins/lookup/url.py

```diff
@@ -26,6 +26,14 @@
     description: Flag to control if the lookup will observe HTTP proxy environment variables when present.
     type: boolean
     default: True
+  url_username:
+    description: Username to use for HTTP authentication.
+    type: string
+    default: ""
```

Comment:
dont set default to "" since you seem to want 'None' below, also consider this PR #42637 , which 'modernizes' this lookup

Reply:
Hi @bcoca ,
How should this be handled with respect to PR #42637? 
Should I modify my PR to include all the changes in PR #42637, as well as update the url_username and url_password to also use get_option? 

I'm pretty new to this git thing so not exactly sure of the mechanics.